src/ext/peek-source.lisp

```diff
@@ -223,6 +228,24 @@
                       (delete-window *source-window*)))
    0))
 
+(define-command grep-help () ()
+  "Show grep help."
+  (with-pop-up-typeout-window (s (make-buffer "*Help*") :erase t)
+    (format s "grep command (M-x grep)~&")
+    (format s "~%")
+    (format s "The left window shows grep results, the right window shows a result in its source file.~&")
+    (format s "~%")
+    (format s "Available keybindings:~&")
+    (format s "- up/down arrows or C-p/C-n: go to the previous/next line~&")
+    (format s "- C-x o or M-o: go to the other window~&")
+    (format s "- Enter: visit the file of the result at point~&")
+    (format s "- Escape or C-x 0: quit~&")
+    (format s "- C-x ?: bring this help~&")
+    (format s "~%")
+    (format s "The results buffer on the left is editable. Any change is written to file and is reflected immediately on the right.~&")
+    (format s "You can use editing tools such as M-x query-replace in the results buffer.~&")
+    (format s "~%")))
```

Comment:
very good 👍 
I am not very good at writing in English, so this kind of thing is very helpful.